Order sensitivity (?) in TestRegisterCallback

From https://circleci.com/gh/cockroachdb/cockroach/773
```
--- FAIL: TestRegisterCallback (0.00s)
	infostore_test.go:588: expected [key1-true key2-true], got [key2-true key1-true]
```

--- a/gossip/infostore_test.go
+++ b/gossip/infostore_test.go
@@ -583,8 +583,9 @@ func TestRegisterCallback(t *testing.T) {
 	wg.Add(2)
 	is.registerCallback("key.*", cb.Add)
 	wg.Wait()
-
-	if expKeys := []string{"key1-true", "key2-true"}; !reflect.DeepEqual(cb.Keys(), expKeys) {
+	actKeys := cb.Keys()
+	sort.Strings(actKeys)
+	if expKeys := []string{"key1-true", "key2-true"}; !reflect.DeepEqual(actKeys, expKeys) {
 		t.Errorf("expected %v, got %v", expKeys, cb.Keys())
 	}
 }

